finality-aleph/src/sync/substrate/verification/cache.rs

```diff
@@ -163,7 +188,36 @@ where
                     + 1,
             ));
         }
+        Ok(())
+    }
 
+    /// Returns the list of Aura authorities for a given block number. Updates cache if necessary.
+    /// This method assumes that the queued Aura authorities will indeed become Aura authorities
+    /// in the next session.
+    pub fn get_aura_authorities(
+        &mut self,
+        number: BlockNumber,
+    ) -> Result<&Vec<AuraId>, CacheError> {
+        let session_id = self.session_info.session_id_from_block_num(number);
+        self.check_and_try_prune(&session_id)?;
+        Ok(match self.aura_authorities.entry(session_id) {
+            Entry::Occupied(occupied) => occupied.into_mut(),
+            Entry::Vacant(vacant) => {
+                let authorities = download_aura_authorities(
+                    &self.authority_provider,
+                    session_id,
+                    &self.session_info,
+                )
+                .ok_or(CacheError::UnknownAuraAuthorities(session_id))?;
+                vacant.insert(authorities)
+            }
+        })
+    }
+
+    /// Returns session verifier for block number if available. Updates cache if necessary.
```

Comment:
Please add something to this comment that it should be called on the block number of the _parent_ of the block being verified – you have this commented below, but this should also be documented here, especially since it's different than above.

Reply:
fixed the comments (not sure if you really meant this one though)